mongodb/collection.go

```diff
@@ -47,9 +47,62 @@ func (c *Collection) Must() *Must {
 	return newMust(c)
 }
 
-// Find returns a Find interface which can be used to either refine the criteria or retrieve a cursor
-func (c *Collection) Find(query interface{}) *Find {
-	return newFind(c.collection, query)
+// Distinct returns the list of distinct values for the given field name in the collection
+func (c *Collection) Distinct(ctx context.Context, filter interface{}, fieldName string) ([]interface{}, error) {
```

Comment:
May I suggest changing the order of the parameters so it matches the [official driver signature](https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#Collection.Distinct)? I believe this could help developers who are familiar with the driver

```suggestion
func (c *Collection) Distinct(ctx context.Context, fieldName string, filter interface{}) ([]interface{}, error) {
```

Reply:
The only thing I don't like about this suggestion is that it then makes Distinct() non uniform in that it is the only method in Collection that takes a filter (or selector) parameter that is not the 2nd parameter in the list (after ctx) - see Count(), Find(), FindOne() etc.

Reply:
Done